src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormHistoricalBalance.js

```diff
@@ -0,0 +1,187 @@
+import React, { useState, useEffect } from 'react'
+import { compose } from 'redux'
+import { graphql } from 'react-apollo'
+import PropTypes from 'prop-types'
+import FormikSelect from '../../../../../components/formik-santiment-ui/FormikSelect'
+import FormikInput from '../../../../../components/formik-santiment-ui/FormikInput'
+import Label from '@santiment-network/ui/Label'
+import {
+  ALL_ERC20_PROJECTS_QUERY,
+  allProjectsForSearchGQL
+} from '../../../../../pages/Projects/allProjectsGQL'
+import { ASSETS_BY_WALLET_QUERY } from '../../../../HistoricalBalance/common/queries'
+import {
+  mapAssetsHeldByAddressToProps,
+  mapToAssets,
+  isPossibleEthAddress
+} from '../../../utils/utils'
+import styles from '../signal/TriggerForm.module.scss'
+
+const isDisabledWalletAddressField = (
+  canUseMappedErc20,
+  target,
+  allErc20Projects
+) => {
+  if (canUseMappedErc20) {
+    return false
+  }
+  if (!target || !target.value) {
```

Comment:
`LOC 25` and `LOC 28` could be merged in one `if` statement. `canUseMappedErc20 || !target || ...`